### PR TITLE
feat: support bytes type in `remote_function`

### PR DESCRIPTION
**Pull request by tswast**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

 🦕

### CODE REVIEW

File: bigframes/functions/remote_function_template.py [hunk 1]
```diff
@@ -126,7 +165,10 @@ def udf_http(request):
         calls = request_json["calls"]
         replies = []
         for call in calls:
-            reply = udf(*call)
+            # TODO: bytes via base64???
```
Comment (tswast) [inline bigframes/functions/remote_function_template.py:168]: Done

```suggestion
```

File: bigframes/functions/remote_function_template.py [hunk 2]
```diff
@@ -65,6 +102,7 @@ def get_pd_series(row):
         "Int64": int,
         "Float64": float,
         "string": str,
+        "binary[pyarrow]": base64.b64decode,
```
Comment (tswast) [inline bigframes/functions/remote_function_template.py:105]: This dictionary worries me. It seems pretty brittle to rely on the string representations of pandas dtypes. We should revisit this in b/345222844

File: tests/system/small/test_remote_function.py [hunk 3]
```diff
@@ -484,17 +485,27 @@ def add_one(x):
 
 
 @pytest.mark.flaky(retries=2, delay=120)
-def test_series_map(session_with_bq_connection, scalars_dfs):
```
Comment (chelsea-lin) [inline tests/system/small/test_remote_function.py:487]: do we still need the old `test_series_map` test?

File: bigframes/dataframe.py [hunk 4]
```diff
@@ -3313,11 +3313,12 @@ def apply(self, func, *, axis=0, args: typing.Tuple = (), **kwargs):
             # Early check whether the dataframe dtypes are currently supported
             # in the remote function
             # NOTE: Keep in sync with the value converters used in the gcf code
-            # generated in generate_cloud_function_main_code in remote_function.py
+            # generated in  in remote_function_template.py
```
Comment (chelsea-lin) [inline bigframes/dataframe.py:3316]: `in generate_cloud_function_main_code`? or just remove extra `in` here?

File: bigframes/functions/remote_function.py [hunk 5]
```diff
@@ -278,26 +289,45 @@ def generate_cloud_function_code(
...
+            is_row_processor=is_row_processor,
         )
         return entry_point
 
     def create_cloud_function(
         self,
         def_,
         cf_name,
+        *,
+        input_types: Tuple[str],
```
Comment (chelsea-lin) [inline bigframes/functions/remote_function.py:305]: Do `input_types` and `output_types` need default values, when they are defined after "*"?

File: bigframes/functions/remote_function_template.py [hunk 6]
```diff
@@ -190,8 +232,22 @@ def generate_udf_code(def_, directory):
     return udf_code_file_name, udf_bytecode_file_name
 
 
-def generate_cloud_function_main_code(def_, directory, is_row_processor=False):
-    """Get main.py code for the cloud function for the given user defined function."""
+def generate_cloud_function_main_code(
+    def_,
+    directory,
+    *,
+    input_types: Tuple[str],
```
Comment (chelsea-lin) [inline bigframes/functions/remote_function_template.py:239]: Similar questions: whether they need default values or not.

**Review by chelsea-lin — APPROVED** [review]
LGTM overall with some minor comments.

File: bigframes/functions/remote_function.py (hunk 5 above)
Comment (tswast) [reply bigframes/functions/remote_function.py:305]: No, the `*` means they are keyword-only. Without a default value they are required.

File: tests/system/small/test_remote_function.py (hunk 3 above)
Comment (tswast) [reply tests/system/small/test_remote_function.py:487]: I think `test_series_map_bytes` should be sufficent. We have other tests that check integer input and output with remote_function.

**Comment by tswast** [comment]
`test_df_apply_axis_1_unsupported_dtype` tests are failing because `binary` is an `ArrowDtype`, so the `isinstance` based validation considers all `ArrowDtype` to be valid.

I will update the validation to check the arrow dtype as well in the case of arrow values.

**Pull request merged**